src/pysoundcomparisons/__main__.py

```diff
@@ -209,90 +209,72 @@ def rename_soundfile(args):
     ffmpeg can be installed via https://www.ffmpeg.org and must be found in a shell call.
     """
 
-    if len(args.args) != 2:
-        args.log.error("need two arguments: old_file_name new_file_name")
-        return
-
-    # OS independent device for suppressing messages while running a shell command
-    nulldev = open(os.devnull, 'w')
-
-    ffmpeg_cmd = 'ffmpeg'
-    if platform.system() == 'Windows':
-        ffmpeg_cmd = 'ffmpeg.exe'
+    ffmpeg_cmd = (platform.system() == 'Windows' and 'ffmpeg.exe') or 'ffmpeg'
 
-    # check if ffmpeg can be called on the local machine
-    if call("%s -version" % (ffmpeg_cmd), stdout=nulldev, stderr=nulldev, shell=True) != 0:
-        nulldev.close()
-        args.log.error("Please make sure that '%s' (https://www.ffmpeg.org) is installed and can be found in a shell call." % (ffmpeg_cmd))
-        return
+    if shutil.which(ffmpeg_cmd) is None:
+        raise OSError("Please make sure that '%s' (https://www.ffmpeg.org) "
+                "is installed and can be found in a shell call." % (ffmpeg_cmd))
 
-    (old_sfname, new_sfname) = args.args
+    try:
+        (old_sfname, new_sfname) = args.args
+    except ValueError:
+        raise ValueError("need two arguments: old_file_name new_file_name")
 
     try:
         new_sfname = SoundfileName(new_sfname)
     except ValueError:
-        nulldev.close()
-        args.log.error("new file name is not valid")
-        return
+        raise ValueError("new file name is not valid")
 
     with _get_catalog(args, 'soundfiles') as catalog:
 
-        if old_sfname in catalog:
+        try:
             obj = catalog.api.get_object(catalog[old_sfname].id)
-        else:
-            nulldev.close()
-            args.log.error("no corresponding entry found for %s" % (old_sfname))
-            return
+        except AttributeError:
```

Comment:
Isn't this supposed to be `KeyError`?

Reply:
No - if `catalog[old_sfname]` is None then `catalog[old_sfname].id` will raise an AttributeError of NoneType

Reply:
Oh. But that's really intransparent behaviour. The `__getitem__` method of `MediaCatalog` really should raise `KeyError` rather than returning `None`.